examples/platforms/posix/radio.c

```diff
@@ -735,44 +783,53 @@ void platformRadioUpdateFdSet(fd_set *aReadFdSet, fd_set *aWriteFdSet, int *aMax
 {
     if (aReadFdSet != NULL && (sState != OT_RADIO_STATE_TRANSMIT || sTxWait))
     {
-        FD_SET(sSockFd, aReadFdSet);
+        FD_SET(sRxFd, aReadFdSet);
 
-        if (aMaxFd != NULL && *aMaxFd < sSockFd)
+        if (aMaxFd != NULL && *aMaxFd < sRxFd)
         {
-            *aMaxFd = sSockFd;
+            *aMaxFd = sRxFd;
         }
     }
 
     if (aWriteFdSet != NULL && platformRadioIsTransmitPending())
     {
-        FD_SET(sSockFd, aWriteFdSet);
+        FD_SET(sTxFd, aWriteFdSet);
 
-        if (aMaxFd != NULL && *aMaxFd < sSockFd)
+        if (aMaxFd != NULL && *aMaxFd < sTxFd)
         {
-            *aMaxFd = sSockFd;
+            *aMaxFd = sTxFd;
         }
     }
 }
 
+// no need to close in virtual time mode.
 void platformRadioDeinit(void)
 {
-    close(sSockFd);
+    if (sRxFd != -1)
+    {
+        close(sRxFd);
```

Comment:
Do we need to leave the multicast group before closing the `sRxFd`?

Reply:
it leaves after the socket is closed.

Reply:
Great